modules/layers/src/path-layer/path-tesselator.js

```diff
@@ -107,59 +110,46 @@ export default class PathTesselator extends Tesselator {
       return;
     }
     const {vertexStart, geometrySize} = context;
+    const p = new Array(3);
 
     // positions   --  A0 A1 B0 B1 B2 B3 B0 B1 B2 --
     // segmentTypes     3  4  4  0  0  0  0  4  4
     for (let i = vertexStart, ptIndex = 0; ptIndex < geometrySize; i++, ptIndex++) {
-      const p = this.getPointOnPath(path, ptIndex);
+      this.getPointOnPath(path, ptIndex, p);
       positions[i * 3] = p[0];
       positions[i * 3 + 1] = p[1];
-      positions[i * 3 + 2] = p[2] || 0;
+      positions[i * 3 + 2] = p[2];
     }
   }
 
   /* Utilities */
   getPathLength(path) {
-    if (Number.isFinite(path[0])) {
-      // flat format
-      return path.length / this.positionSize;
-    }
-    return path.length;
+    return path.length / this.positionSize;
   }
 
-  getPointOnPath(path, index) {
-    if (Number.isFinite(path[0])) {
-      // flat format
-      const {positionSize} = this;
-      if (index * positionSize >= path.length) {
-        // loop
-        index += 1 - path.length / positionSize;
-      }
-      // TODO - avoid creating new arrays when using binary
-      return [
-        path[index * positionSize],
-        path[index * positionSize + 1],
-        positionSize === 3 ? path[index * positionSize + 2] : 0
-      ];
-    }
-    if (index >= path.length) {
+  getPointOnPath(path, index, target = []) {
```

Comment:
could we get a one line comment header in front of each method here hinting a some of the key points of the work being done in each method.